### PR TITLE
Added some liitle features I personally like

### PR DESCRIPTION
**Pull request by DanielVolz**
Main changes:

- Show/hide hover card in overlay
- Show/hide draw cards odds in overlay
- Resize hover card in overlay and in collection

I'm not a experienced programmer, so there might be some bad code in there.

### CODE REVIEW

**Comment by Manuel-777** [comment]
Hi Daniel! Thanks for contributing!

I am merging #475 today/tomorrow, which makes some big changes to tge overlays, I would love to bring these to it as well!

**Comment by DanielVolz** [comment]
Oh, the new overlays look very sweat :)
Thank you for making the best MTGA Tracker and for keeping it open source!

**Comment by DanielVolz** [comment]
I merged the new overlay v3 and removed the hide odds feature because there is an overlay for that and I didn't notice it till today ¯\_(ツ)_/¯.

**Comment by DanielVolz** [comment]
- Fixed resizing issues with overlays. There was a bug where you couldn't resize am overlay if it was big enough to touch the top and bottom edge.
- Swapped next/previous arrows for odds sample.

File: window_main/settings.js [hunk 1]
```diff
@@ -660,6 +660,22 @@ function appendOverlay(section) {
       updateUserSettings,
       OVERLAY_DRAFT_MODES.includes(settings.mode)
     );
+    addCheckbox(
+      overlaySection,
+      "Show draw crads odds",
```
Comment (lusbenjamin) [inline window_main/settings.js]: spelling nitpick: "Show draw *cards* odds"
or maybe just: "Show odds"

File: window_main/settings.js [hunk 2]
```diff
@@ -660,6 +660,22 @@ function appendOverlay(section) {
...
+      `overlay_${index}_draw_odds`,
+      settings.draw_odds,
+      updateUserSettings,
+      OVERLAY_DRAFT_MODES.includes(settings.mode)
+    );
+    addCheckbox(
+      overlaySection,
+      "Show cards on hoover",
```
Comment (lusbenjamin) [inline window_main/settings.js]: spelling nitpick: "Show cards on *hover*"

File: window_main/settings.js [hunk 3]
```diff
@@ -660,6 +660,22 @@ function appendOverlay(section) {
...
+      `overlay_${index}_cards_overlay`,
+      settings.cards_overlay,
+      updateUserSettings,
+      OVERLAY_DRAFT_MODES.includes(settings.mode)
```
Comment (lusbenjamin) [inline window_main/settings.js]: major concern: the expression `OVERLAY_DRAFT_MODES.includes(settings.mode)` controls which overlays *cannot* use the feature. How do you intend for this to work with draft overlays?

Currently, draft overlays also use the card hover feature. For example on draft pack 2-pick 1, the player may wish to review all picks from pack 1. This requires clicking back through the draft overlay picks and hovering over those cards to think about the partially built deck.

File: window_overlay/index.css [hunk 4]
```diff
@@ -61,6 +61,7 @@ body {
 .main_hover {
     margin-left: auto !important;
     margin-right: auto !important;
+    margin-bottom: 70px !important;
```
Comment (lusbenjamin) [inline window_overlay/index.css]: this entire file is deprecated (see #530)

File: window_overlay/overlay.js [hunk 5]
```diff
@@ -332,10 +332,20 @@ ipc.on("settings_updated", (_event, index) => {
   $$(".overlay_decklist")[0].innerHTML = "";
   $$(".overlay_decklist")[0].style.display = settings.deck ? "" : "none";
 
+  $$(".main_hover")[0].style.width = pd.cardsSizeOverlayCard + "px";
+  $$(".main_hover")[0].style.height =
+    pd.cardsSizeOverlayCard / 0.71808510638 + "px";
+
```
Comment (lusbenjamin) [inline window_overlay/overlay.js]: this entire file is deprecated (see #530)

File: window_overlay_v3/overlay.js [hunk 6]
```diff
@@ -270,11 +277,20 @@ function settingsUpdated() {
...
+    queryElements(mainHoverDom)[0].style.width = pd.cardsSizeOverlayCard + "px";
+    queryElements(mainHoverDom)[0].style.height =
+      pd.cardsSizeOverlayCard / 0.71808510638 + "px";
 
     const showClock =
       _overlay.clock && !OVERLAY_DRAFT_MODES.includes(_overlay.mode);
     queryElements(clockDom)[0].style.display = showClock ? "" : "none";
 
+    const showCardOverlay =
+      _overlay.cards_overlay && !OVERLAY_DRAFT_MODES.includes(_overlay.mode);
```
Comment (lusbenjamin) [inline window_overlay_v3/overlay.js]: Does this logic mean that this PR no longer supports the card hover feature for draft overlays?

File: window_overlay_v3/index.html [hunk 7]
```diff
@@ -25,7 +25,11 @@
             <div id="overlay_4" class="overlay_container click-on"></div>
             <div id="overlay_5" class="overlay_container click-on"></div>
             <div class="overlay_hover_container">
-                <img class="main_hover" style="opacity: 0;"></img>
+                <img id="hover_1" class="main_hover" style="opacity: 0;"></img>
+                <img id="hover_2" class="main_hover" style="opacity: 0;"></img>
+                <img id="hover_3" class="main_hover" style="opacity: 0;"></img>
+                <img id="hover_4" class="main_hover" style="opacity: 0;"></img>
+                <img id="hover_5" class="main_hover" style="opacity: 0;"></img>
```
Comment (lusbenjamin) [inline window_overlay_v3/index.html]: Just curious: why do we need one of these for each overlay instead of re-using the same one like before?

**Comment by DanielVolz** [comment]
Hi @lusbenjamin , thanks for reviewing my PR. I have to admit, I'm not an experienced coder and this is kind of my first PR, so please excuse me my mistakes :)
I'm not familiar with this process but I wanted to contribute to this wonderful program!

And you are right about every problem, I might have PRed this too soon. But I will try to correct all my mistakes. The problem with draft is, I don't play it but I think I will give it a go, at least for debugging purposes :)

File: window_overlay_v3/index.html (hunk 7 above)
Comment (DanielVolz) [reply window_overlay_v3/index.html]: Because you might turn the feature on/off for specific overlays.

File: window_overlay_v3/overlay.js (hunk 6 above)
Comment (DanielVolz) [reply window_overlay_v3/overlay.js]: I have to look into draft and fix this.

File: window_overlay/overlay.js (hunk 5 above)
Comment (DanielVolz) [reply window_overlay/overlay.js]: I added it the new v3 overlay.

File: window_overlay/index.css (hunk 4 above)
Comment (DanielVolz) [reply window_overlay/index.css]: I added it the new v3 overlay. I hope ;)

File: window_main/settings.js (hunk 3 above)
Comment (DanielVolz) [reply window_main/settings.js]: I need to test this in draft.

File: window_main/settings.js (hunk 2 above)
Comment (DanielVolz) [reply window_main/settings.js]: 👍

File: window_main/settings.js (hunk 1 above)
Comment (DanielVolz) [reply window_main/settings.js]: 👍

File: window_main/settings.js (hunk 2 above)
Comment (Manuel-777) [reply window_main/settings.js]: A *dam mistake* .. :V

Comment (DanielVolz) [reply window_main/settings.js]: Herbert likes this this comment. <3

**Comment by DanielVolz** [comment]
>     * ❔ confused by why we need the new "show card odds" option

This made more sens with the old overlay and the old hover card position. The cards draw odds kind of weren't pleasant to my eyes. 

But I agree, it's a minor feature but personally I like it. I like to use the "Library and Odds" overlay because you can see the individual card odds but like I said, I don't like the draw card odds.

test

File: window_main/settings.js (hunk 3 above)
Comment (DanielVolz) [reply window_main/settings.js]: I would say it is ok to allow this feature for every overlay.

File: window_overlay_v3/overlay.js [hunk 8]
```diff
@@ -617,7 +633,10 @@ function updateMatchView(index) {
     });
   }
 
-  if (overlayMode === OVERLAY_ODDS || overlayMode === OVERLAY_MIXED) {
+  if (
+    overlayMode === OVERLAY_ODDS ||
+    (overlayMode === OVERLAY_MIXED && settings.draw_odds)
```
Comment (lusbenjamin) [inline window_overlay_v3/overlay.js]: if we want this setting to control the stats panel for both modes where it appears, we might want:
```
if (
  (overlayMode === OVERLAY_ODDS || overlayMode === OVERLAY_MIXED) && settings.draw_odds
) {
...
}
```

File: window_main/settings.js [hunk 9]
```diff
@@ -660,6 +660,22 @@ function appendOverlay(section) {
       updateUserSettings,
       OVERLAY_DRAFT_MODES.includes(settings.mode)
     );
+    addCheckbox(
+      overlaySection,
+      "Show draw crads odds",
+      `overlay_${index}_draw_odds`,
+      settings.draw_odds,
+      updateUserSettings,
+      OVERLAY_DRAFT_MODES.includes(settings.mode)
```
Comment (lusbenjamin) [inline window_main/settings.js]: I think we want this setting to only be enabled (not grayed out) for the card odds and mixed modes. Does this logic work?
`!(settings.mode === OVERLAY_ODDS || settings.mode === OVERLAY_MIXED)`

**Comment by lusbenjamin** [comment]
P.S. - sorry for causing merge conflicts for you several times. If you let @Manuel-777 know your Discord tag, he can give you access to the #devs channel to discuss features and builds

**Comment by DanielVolz** [comment]
@lusbenjamin I think I added all the suggestions you've mentioned.
Sorry, for the last redundant merge. I'm still figuring out how the hell git works.

My name on the discord is hallerharry23.

**Pull request merged**